FerramAerospaceResearch/FARAeroComponents/VehicleOcclusion.cs

```diff
@@ -0,0 +1,828 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using FerramAerospaceResearch.UnityJobs;
+using Unity.Collections;
+using Unity.Jobs;
+using Unity.Mathematics;
+using UnityEngine;
+using UnityEngine.Jobs;
+using UnityEngine.Profiling;
+using FerramAerospaceResearch.Settings;
+
+/* Theory of Job Control:
+ * At beginning of FixedUpdate cycle, pre-calculate the occlusion orientations we need.
+ * Convection occlusion calcuation and sun occlusion calculation are separate orientations.
+ * Weighted-average / lerp the 3 closest orientations; this can be 6 raycasts on first execution if we missed all caches.
+ * On exemplar system (4-core I5-6500 3.2GHz 16GB RAM):
+ *   <2ms at FixedUpdate.ObscenelyEarly to gather the raycast data
+ *  Expensive raycasts can cost ~4ms (10,000 rays), cheaper ones 0.5ms
+ *
+ * */
+namespace FerramAerospaceResearch.FARAeroComponents
+{
+    public class VehicleOcclusion : MonoBehaviour
```

Comment:
Is there a reason this is a `MonoBehaviour` and not `VesselModule`? A module can be found without using `GetComponent` which has to call a native method. `FARVesselAero` is basically never used and it's lifetime is bound to `Vessel` anyway. Removing the coupling between the two would reduce complexity

Reply:
Some reason.  Yes, I could make it a VesselModule.  I don't think that is more efficient than using Unity's GetComponent method, since that's the normal way Unity expects things.  However, I cannot completely decouple from FARVesselAero because we require a notification every time voxelization completes.  Maybe I could register with something else and get that notification another way?  But that seemed the right kind of place.  So some coupling has to remain, though becoming a VesselModule would remove the need to get created in FARVesselAero.

VehicleOcclusion doesn't need to survive outside of a Vessel, either.